main_penalty_calculate/builders/violators_avaliations_builder.py

```diff
@@ -1,77 +1,137 @@
+from datetime import datetime
+
 from ..models import ViolatorAvaliation, IdentityCard
 
 
 class ViolatorsAvaliationsBuilder:
     _LICENSE_PLATE_IN_LIST = 0
     _FIRST_TRAFFIC_VIOLATION = 0
+    _DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+    _VALIDITY_PERIOD_OF_INFRINGEMENT = 30
+    _INVALID_DEMERIT_POINTS = 0
+    _INFRACTION_PENALTIES = {
+        'Leve': 3,
+        'Média': 4,
+        'Grave': 5,
+        'Gravíssima': 7
+    }
 
     def __init__(self, traffic_violations):
         self._traffic_violations = traffic_violations
+        self._violators_avaliations = []
 
     def _is_identity_card_number_already_present(
         self,
-        identity_card_number_to_be_checked,
-        identity_card_number_list
+        identity_card_in_review
     ):
-        for identity_card in identity_card_number_list:
+        for identity_card in self._violators_avaliations:
             if (
                 identity_card.identity_card_number ==
-                identity_card_number_to_be_checked
+                identity_card_in_review
             ):
                 return True
         return False
 
     def _is_license_plate_number_already_present(
         self,
-        license_plate_number_to_be_checked,
-        license_plate_list
+        license_plate_list,
+        license_plate_in_review
     ):
         for license_plate_number in license_plate_list:
             if (
                 license_plate_number ==
-                license_plate_number_to_be_checked
+                license_plate_in_review
             ):
                 return True
         return False
 
-    def _agroup_license_plates(
+    def _aggregate_license_plates(
         self,
-        license_plate_in_review,
-        violators_list,
-        violator_in_review
+        violator_registered,
+        license_plate_in_review
     ):
-        for violator in violators_list:
-            if violator.identity_card_number == violator_in_review:
-                if not self._is_license_plate_number_already_present(
-                    license_plate_in_review,
-                    violator.license_plate_numbers
-                ):
-                    violator.license_plate_numbers.append(
-                        license_plate_in_review
-                    )
+        if not self._is_license_plate_number_already_present(
+            violator_registered.license_plate_numbers,
+            license_plate_in_review
+        ):
+            violator_registered.license_plate_numbers.append(
+                license_plate_in_review
+            )
+
+    def _aggregate_demerit_points(
+        self,
+        violator_registered,
+        notification_date_in_review,
+        infraction_date_in_review,
+        type_infraction_in_review
```

Comment:
Existe algum motivo para troca de nomes das variaveis?
Nao sei se tah claro o que eh esses 4 parametros, o que `in review`?

Reply:
O motivo era para se assemelhar com os outros parâmetros em revisão... Mas de fato eu também achei estranho assim que mudei kkkkkkkk. Voltarei pra forma anterior!